src/AlgebraicGeometry/Schemes/AffineAlgebraicSet/Objects/Types.jl

```diff
@@ -1,6 +1,7 @@
 @attributes mutable struct AffineAlgebraicSet{BaseRing<:Field, RingType<:MPolyAnyRing} <: AbsAffineAlgebraicSet{BaseRing, RingType}
   X::Spec
   function AffineAlgebraicSet(X::Spec; check::Bool=true)
+    @check
     if check
       is_geometrically_reduced(X) || error("algebraic sets must be geometrically reduced")
     else
```

Comment:
Another opportunity to avoid the "stand-alone" `@check`:
```suggestion
    @check is_geometrically_reduced(X) "algebraic sets must be geometrically reduced"
```
(and then adjust the following code, too)